## Summary of the dataset:

In [240]:
import pandas as pd
df = pd.read_csv("dataset/Housing.csv")

**Data Info:**


In [243]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 545 entries, 0 to 544
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   price             545 non-null    int64 
 1   area              545 non-null    int64 
 2   bedrooms          545 non-null    int64 
 3   bathrooms         545 non-null    int64 
 4   stories           545 non-null    int64 
 5   mainroad          545 non-null    object
 6   guestroom         545 non-null    object
 7   basement          545 non-null    object
 8   hotwaterheating   545 non-null    object
 9   airconditioning   545 non-null    object
 10  parking           545 non-null    int64 
 11  prefarea          545 non-null    object
 12  furnishingstatus  545 non-null    object
dtypes: int64(6), object(7)
memory usage: 55.5+ KB


**Numerical Summary:**


In [246]:
 df.describe()

price          area    bedrooms   bathrooms     stories  \
count  5.450000e+02    545.000000  545.000000  545.000000  545.000000   
mean   4.766729e+06   5150.541284    2.965138    1.286239    1.805505   
std    1.870440e+06   2170.141023    0.738064    0.502470    0.867492   
min    1.750000e+06   1650.000000    1.000000    1.000000    1.000000   
25%    3.430000e+06   3600.000000    2.000000    1.000000    1.000000   
50%    4.340000e+06   4600.000000    3.000000    1.000000    2.000000   
75%    5.740000e+06   6360.000000    3.000000    2.000000    2.000000   
max    1.330000e+07  16200.000000    6.000000    4.000000    4.000000   

          parking  
count  545.000000  
mean     0.693578  
std      0.861586  
min      0.000000  
25%      0.000000  
50%      0.000000  
75%      1.000000  
max      3.000000

By analyzing the statistical measures of the dataset's attributes, we gathered the following insights:

**Price:** The price of houses ranges **from 1,750,000 to 13,300,000**, with a median of **4,340,000**. This indicates a skew towards higher prices, as the 3rd quartile is much closer to the median than the maximum value, suggesting potential outliers at the higher end.

**Area:** House sizes vary **from 1,650 to 16,200** square feet, with a median of **4,600** square feet. This suggests that there are more larger houses compared to smaller ones in the dataset.

**Bedrooms:** The number of bedrooms ranges **from 1 to 6**, with a median of **3**, this implies that most houses have three or fewer bedrooms, with fewer homes having more than that.

**Bathrooms:** The dataset shows a range of **1 to 4** bathrooms per house, with a median of **1**, indicating that most homes have just one bathroom.

**Stories:** Houses in this dataset have **between 1 and 4** stories, with a median of **2**, meaning that the majority of homes have two or fewer stories.

**Parking:** The number of parking spots ranges **from 0 to 3**, with a median of **0**. This suggests that most houses in the dataset likely do not have parking facilities.

**Variance:**

indicates how dispersion and spread of values.Therefore, our variance results indicate:

price have high variance, so the level of dispersion and spread of values is high.

stories have low variance, so the level of dispersion and spread of values is low.

In [250]:
df.var(numeric_only=True)

price        3.498544e+12
area         4.709512e+06
bedrooms     5.447383e-01
bathrooms    2.524757e-01
stories      7.525432e-01
parking      7.423300e-01
dtype: float64

**Categorical Summary:**

In [253]:
df.describe(include=[object])

mainroad guestroom basement hotwaterheating airconditioning prefarea  \
count       545       545      545             545             545      545   
unique        2         2        2               2               2        2   
top         yes        no       no              no              no       no   
freq        468       448      354             520             373      417   

       furnishingstatus  
count               545  
unique                3  
top      semi-furnished  
freq                227

By analyzing dataset's Categorical attributes, we gathered the following insights:

**mainroad:** The majority of the houses in this dataset are **located on a main road**, with **468 out of 545** properties having this feature. This shows that being near a main road is **common** in this dataset, likely offering convenience and accessibility.

**guestroom:** Most houses do not have a guestroom, as **448** properties show **no** for this feature. This might suggest that having a guestroom is considered more of a **luxury or less common** in this set of houses.

**basement:** Basements are also **less common**, with only **354** properties lacking one. This could be influenced by regional preferences or property type.

**hotwaterheating:** The vast majority of homes do not have hot water heating, with **520** properties showing **no**. This suggests that either the homes use alternative heating methods or that hot water heating is **less of a priority** in this dataset.

**airconditioning:** Air conditioning isn't widely present, with **373** homes **not having** it. This might reflect the climate of the area or the age of the properties, where air conditioning isn't a standard feature.

**prefarea:** Only **128** homes are located in a **preferred area**, as most **417** homes are not. This could suggest a broader variety of locations and property types, with the majority not being classified as premium locations.

**furnishingstatus:** Furnishing is a mix, but the **most common** option is **semi-furnished**, with **227** homes in this category. This could appeal to buyers looking for flexibility between a fully furnished home and an empty one.

**Missing Values:**
there are no missing values

In [257]:
df.isnull().sum()

price               0
area                0
bedrooms            0
bathrooms           0
stories             0
mainroad            0
guestroom           0
basement            0
hotwaterheating     0
airconditioning     0
parking             0
prefarea            0
furnishingstatus    0
dtype: int64

## Data Visualization:

## 1. Scatter plot:
A scatter plot is a useful visualization tool that helps us understand the **relationship between two numerical attributes**. It displays data points as dots on a **two-dimensional** graph, with one attribute on the x-axis and the other on the y-axis. This format makes it easy to identify trends, patterns, and correlations, transforming complex data into a more accessible format. Scatter plots are widely used in various fields, making data analysis simpler and more intuitive. 

In [ ]:
import matplotlib.pyplot as plt
import seaborn as sns

##plt.scatter( df['area'], df['price'],c=df['price'])+plt.colorbar(label='Price')

plt.figure(figsize=(6, 4))
sns.scatterplot(x='area', y='price', data= df)
plt.title('Scatter Plot of Area vs Price')
plt.xlabel('Area')
plt.ylabel('Price') 
plt.show()